### PR TITLE
#2527 Allow collecting of host stats within docker containers by enab…

### PR DESCRIPTION
**Pull request by MatthewCh**
Added fix for #2527 

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by danielnelson** [comment]
If we are going to use an environment variable we should use HOST_PROC and HOST_SYS to match gopsutil, though I'd prefer to use options on the plugins.

**Comment by MatthewCh** [comment]
Agreed.

But I wasn't sure if these were the only 2 plugins that could take these overrides, if not it would be easier to use environment variables so we wouldn't have to redefine the same options on multiple plugins.

But I will make the proposed change since it doesn't seem like that is the case.

**Comment by danielnelson** [comment]
It is a good point about wanting to share these, it could be good to have support for either or even both methods.  If you do the environment variables though lets just use the same names as gopsutil since other users are already setting it for other plugins https://github.com/influxdata/telegraf/issues/1544#issuecomment-278955406

**Comment by MatthewCh** [comment]
Thanks for now I will have the change only for environment variables, and I have changed the names to match gopsutil.

**Comment by danielnelson** [comment]
In order to match gopsutil, the `HOST_SYS` variable should point to the `/sys` location and `HOST_PROC` should point to the `/proc` location.  All of the variables are documented here: https://github.com/shirou/gopsutil#usage

**Comment by MatthewCh** [comment]
Thanks, understood. I have made the change and I will retest it tomorrow. If you could take a quick look that would be helpful, i'm still relatively new to go.

File: plugins/inputs/system/linux_sysctl_fs.go [hunk 1]
```diff
@@ -23,6 +24,9 @@ func (_ SysctlFS) Description() string {
 func (_ SysctlFS) SampleConfig() string {
 	return sysctlFSSampleConfig
 }
+func OSGetEnv(key string) string {
```
Comment (danielnelson) [inline plugins/inputs/system/linux_sysctl_fs.go]: These functions don't really encapsulate anything, you could remove them or maybe you could have them return the default value when unset and call it something like `GetHostProc() string` or `GetHostSys() string`.

File: plugins/inputs/system/linux_sysctl_fs.go [hunk 2]
```diff
@@ -80,9 +84,14 @@ func (sfs *SysctlFS) Gather(acc telegraf.Accumulator) error {
...
 func init() {
+	sysPath := "/proc/sys"
+	if OSGetEnv("HOST_SYS") != "" {
+		sysPath = OSGetEnv("HOST_SYS")
+	}
+
 	inputs.Add("linux_sysctl_fs", func() telegraf.Input {
 		return &SysctlFS{
-			path: "/proc/sys/fs",
+			path: sysPath + "/fs",
```
Comment (danielnelson) [inline plugins/inputs/system/linux_sysctl_fs.go]: Use `path.Join` for all of the paths

**Comment by MatthewCh** [comment]
Thanks that helped. I have made the appropriate changes.

File: plugins/inputs/system/linux_sysctl_fs.go [hunk 3]
```diff
@@ -80,9 +89,10 @@ func (sfs *SysctlFS) Gather(acc telegraf.Accumulator) error {
 }
 
 func init() {
+
 	inputs.Add("linux_sysctl_fs", func() telegraf.Input {
 		return &SysctlFS{
-			path: "/proc/sys/fs",
+			path: path.Join(GetHostSys(), "/fs"),
```
Comment (danielnelson) [inline plugins/inputs/system/linux_sysctl_fs.go]: I guess this should actually use GetHostProc too, since `/proc/sys` and `/sys` are not the same.

Comment (MatthewCh) [inline plugins/inputs/system/linux_sysctl_fs.go]: Are you saying it should be more like path.Join(GetHostProc(), "/sys/fs")?

I am not sure would HOST_SYS be of any use then?

Comment (danielnelson) [inline plugins/inputs/system/linux_sysctl_fs.go]: Yes, looks like we are not actually using HOST_SYS here.  I think the cgroups plugin is an example where it is needed.

**Comment by MatthewCh** [comment]
Cool, I updated it, but I am unable to find where cgroup uses the /sys path.

**Pull request merged**